src/VirtoCommerce.CatalogModule.Web/Controllers/Api/CatalogModuleAssociationsController.cs

```diff
@@ -0,0 +1,99 @@
+        /// </summary>
+        /// <remarks>Returns list of associations for specified product</remarks>
+        /// <param name="productid">Owner product id</param>
+        [HttpGet]
+        [Route("")]
+        [Authorize(ModuleConstants.Security.Permissions.Read)]
+        public async Task<ActionResult<ProductAssociation[]>> GetAllAssociations(string productid)
+        {
+            var result = await _associationService.GetAssociationsAsync(new string[] { productid });
+            return Ok(result);
+        }
+
+        /// <summary>
+        /// Updates the specified association
+        /// </summary>
+        /// <remarks>Updates the specified association</remarks>
+        /// <param name="productid">Owner product id</param>
+        /// <param name="association">The association</param>
+        [HttpPost]
+        [Route("")]
+        [Authorize(ModuleConstants.Security.Permissions.Update)]
+        public async Task<ActionResult> UpdateAssociation(string productid, [FromBody] ProductAssociation association)
+        {
+            await _associationService.UpdateAssociationAsync(productid, association);
+            return Ok();
+        }
+
+        /// <summary>
+        /// Updates set of associations for specified product
+        /// </summary>
+        /// <remarks>Updates the specified association</remarks>
+        /// <param name="productid">Owner product id</param>
+        /// <param name="associations">Associations</param>
+        [HttpPost]
+        [Route("set")]
+        [Authorize(ModuleConstants.Security.Permissions.Update)]
+        public async Task<ActionResult> UpdateAssociationSet(string productid, [FromBody] ProductAssociation[] associations)
+        {
+            await _associationService.UpdateAssociationSetAsync(productid, associations);
+            return Ok();
+        }
+
+        /// <summary>
+        /// Deletes specified associations
+        /// </summary>
+        /// <remarks>Updates the specified association</remarks>
+        /// <param name="productid">Owner product id</param>
+        /// <param name="ids">associations to delete ids</param>
+        [HttpDelete]
+        [Route("")]
+        [Authorize(ModuleConstants.Security.Permissions.Delete)]
+        public async Task<ActionResult> Delete(string productid, [FromBody]string[] ids)
+        {
+            await _associationService.DeleteAssociationAsync(ids);
+            return Ok(HttpStatusCode.NoContent);
+        }
+
+        /// <summary>
+        /// Returns associations by search criteria
+        /// </summary>
+        /// <remarks>Returns associations by search criteria</remarks>
+        /// <param name="productid">Owner product id</param>
+        [HttpPost]
+        [Route("search")]
+        [Authorize(ModuleConstants.Security.Permissions.Read)]
+        public async Task<ActionResult<ProductAssociation[]>> Search(string productid, [FromBody]ProductAssociationSearchCriteria criteria)
```

Comment:
We do not need productId here, they could be searched by array of ids specified in ProductAssociationSearchCriteria. So need to remove param and change routing